### PR TITLE
Refactor EditOnGitHub

### PR DESCRIPTION
**Pull request by B2o5T**
see https://github.com/the-guild-org/the-guild-docs/pull/579

### CODE REVIEW

**Comment by B2o5T** [comment]
It's ❌ as I skipped canary

File: .changeset/long-mails-serve.md [hunk 1]
```diff
@@ -4,40 +4,33 @@
...
+      <DocsContent>{content}</DocsContent>
+      <DocsTOC>
+        <TOC />
+        <BottomNavigation />
+        <EditOnGitHubButton
+          repo="the-guild-org/the-guild-docs"
+          baseDir="examples/basic"
+          branch="main"
+          sourceFilePath={sourceFilePath}
+        />
```
[the remark this answers is not in the document]
Comment (n1ru4l) [reply .changeset/long-mails-serve.md:31]: Isn't this kind of contradicting with how giscus is configured? I don't really mind, I just wanna ask.

File: packages/client/src/EditOnGitHubButton.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React, { FC } from 'react';
+import { Button } from '@chakra-ui/react';
+
+const GitHubIcon: FC<{ className?: string; size?: number }> = ({ className, size = 24 }) => {
```
Comment (n1ru4l) [inline packages/client/src/EditOnGitHubButton.tsx:4]: https://fettblog.eu/typescript-react-why-i-dont-use-react-fc/

Comment (n1ru4l) [reply packages/client/src/EditOnGitHubButton.tsx:4]:
```suggestion
const GitHubIcon = ({ className, size = 24 }: { className?: string; size?: number }) => {
```

File: packages/client/src/giscus.tsx [hunk 3]
```diff
@@ -15,16 +12,14 @@ export interface GiscusProps {
 
 let GiscusKeyInc = 0;
 
-export function Giscus({ category, categoryId, repo, repoId }: GiscusProps) {
+const Giscus: FC<GiscusProps> = ({ category, categoryId, repo, repoId }) => {
```
Comment (n1ru4l) [inline packages/client/src/giscus.tsx:15]:
```suggestion
const Giscus = ({ category, categoryId, repo, repoId }: GiscusProps) => {
```

File: packages/client/src/giscus.tsx [hunk 4]
```diff
@@ -15,16 +12,14 @@ export interface GiscusProps {
...
 
   const dataTheme = useColorModeValue('light', 'dark');
 
-  const key = useMemo(() => {
-    return `${dataTheme}${asPath}${++GiscusKeyInc}`.replace(/\//g, '_');
-  }, [dataTheme, asPath]);
+  const key = useMemo(() => `${dataTheme}${asPath}${++GiscusKeyInc}`.replace(/\//g, '_'), [dataTheme, asPath]);
 
-  const isBrowser = useIsBrowserSSRSafe();
+  const isBrowser = typeof window !== 'undefined';
```
Comment (n1ru4l) [inline packages/client/src/giscus.tsx:22]:
```suggestion
  const isBrowser = !!globalThis.window;
```

File: packages/client/src/EditOnGitHubButton.tsx (hunk 2 above)
Comment (B2o5T) [reply packages/client/src/EditOnGitHubButton.tsx:4]: but now this fn don't return ReactElement :)

Comment (n1ru4l) [reply packages/client/src/EditOnGitHubButton.tsx:4]: Its inferred, but you can also add it explicitly.

```suggestion
const GitHubIcon = ({ className, size = 24 }: { className?: string; size?: number }): React.ReactElement => {
```

File: packages/client/src/giscus.tsx (hunk 3 above)
Comment (B2o5T) [reply packages/client/src/giscus.tsx:15]: We can discuss of forbidding `FC`, `FunctionalComponent` imports from react in our future guild config ;)

File: .changeset/long-mails-serve.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (B2o5T) [reply .changeset/long-mails-serve.md:31]: I think even Giscus can be simplified by exporting as simple component

[the remark this answers is not in the document]
Comment (n1ru4l) [reply .changeset/long-mails-serve.md:31]: agreed, maybe we can change it in the next major!

The only benefit I see here is hiding the `sourceFilePath` prop passing task from the user. But it is nothing we really have to do.

**Review by n1ru4l — APPROVED** [review]

**Pull request merged**